src/duckling/selection/selection.service.ts

```diff
@@ -23,82 +28,137 @@ export interface Selection {
 @Injectable()
 export class SelectionService {
 
-    public selection : BehaviorSubject<Selection>;
+    public selections : BehaviorSubject<Selection[]>;
 
     constructor(private _store : StoreService,
                 private _entitySystem : EntitySystemService,
+                private _drawerService : EntityDrawerService,
                 private _layerService : EntityLayerService,
-                private _drawerService : EntityDrawerService) {
-        this.selection = new BehaviorSubject({});
+                private _entityBoxService : EntityBoxService,
+                private _renderPriority: RenderPriorityService) {
+        this.selections = new BehaviorSubject([]);
         this._store.state.subscribe(state => {
-            this.selection.next(this._getSelection(state.selection.selectedEntity))
+            this.selections.next(this._getSelections(state.selections.selectedEntities));
         });
     }
 
     /**
      * Select the entity.
-     * @param  entityKey The key of the entity to select.
-     * @param  mergeKey  Optional mergeKey, describes which commands the action should be merged with.
+     * @param  entityKeys The keys of entities to select
+     * @param  mergeKey   Optional mergeKey, describes which commands the action should be merged with.
      */
-    select(entityKey : EntityKey, mergeKey? : any) {
-        this._store.dispatch(_selectionAction(entityKey), mergeKey);
+    select(entityKeys : EntityKey[] = [], mergeKey? : any) {
+        this._store.dispatch(_selectionAction(entityKeys), mergeKey);
     }
 
     /**
      * Clear the current selection.
      * @param  mergeKey  Optional mergeKey, describes which commands the action should be merged with.
      */
     deselect(mergeKey? : any) {
-        this.select("", mergeKey);
+        this.select([], mergeKey);
     }
 
-    private _getSelection(selectedEntity : EntityKey) : Selection {
-        if (!selectedEntity) {
-            return {};
+    getEntityKeyAtPosition(position: Vector) : EntityKey {
+        let entities = Array.from(this._renderPriority.sortEntities(this._entitySystem.entitySystem.getValue()));
+        entities.reverse();
+        let taggedEntity = entities
+            .filter(entity => this._drawerService.isEntityVisible(entity.entity))
+            .find(entity => this._entityContainsPoint(entity.key, position));
+
+        if (taggedEntity) {
+            return taggedEntity.key;
+        } else {
+            return null;
         }
+    }
+
+    getEntityKeysAtArea(position : Vector, dimension : Vector) : EntityKey[] {
```

Comment:
renamed to getEntityKeysInSelection, and renamed the parameters to selectionPosition and selectionDimension